(DOCSP-44848) Migration page round two (#192)
mongodb/docs-atlas-architecture: source/migration.txt

.. _arch-center-migration:

=========
Migration
=========

.. default-domain:: mongodb

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: atlas architecture center
   :description: Learn about live migrating data into Atlas from on-premises MongoDB databases using several automated processes.

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: onecol

You can migrate data from your on-premises MongoDB deployments to |service| using one of 

> a variety of methods. We recommend using :atlas:`|service| live migration </atlas/import/#migrate-or-import-data>` 

Reviewer: ```suggestion
a variety of methods. We recommend using |service| :atlas:`live migration </atlas/import/#migrate-or-import-data>` 
```
I'm fairly certain the substitution won't render properly inside of a role

when possible because it automates many of the tasks with the least amount of downtime, but you 
can use other tools that accommodate the variety and complexity inherent to database migration. 

Live Migration Overview
----------------------- 

|service| live migration automates moving data from on-premises MongoDB databases to |service|.

> You can pull data from an on-premises MongoDB database or push data using |com|, 

Reviewer: ```suggestion
You can pull data from an on-premises MongoDB database or push data using |com|, 
```

@JuliaMongo wasn't push migration going to be removed soon? perhaps I've got something mixed up

Author: @JuliaMongo will address in a separate PR if there's any lead on this

but with either method, |service| live migration includes the following features:

> - The migration host always encrypts traffic to the |service| {+cluster+}. To encrypt 

Reviewer: we can stop using the {+cluster+} source constant. you can make this change throughout the PR
```suggestion
- The migration host always encrypts traffic to the |service| cluster. To encrypt 
```

- Detailed instructions help you provision migration hosts and scale destination {+clusters+} 
  to control costs. Recommendations include appropriate {+cluster+} sizing 
  and temporary scaling, followed by resizing to optimal levels post-migration.

- Live migration uses |mongosync| to facilitate fast cutover through parallel data copying.
  Processes manage temporary network interruptions and {+cluster+} elections, using continuous 
  data synchronization and a final cutover phase to achieve minimal downtime. Retry 
  mechanisms and pre-migration validations enhance resilience against interruptions.

- Monitor migrations with real-time status updates and notifications.

Live Migration Methods
----------------------

You can use |com| to push data into |service| or use a live migration server 
to pull data into |service|.

Ensure you allocate adequate |cpu| and network resources for the migration host. 
While you can run multiple concurrent migrations, each deployment must have a 
dedicated migration host.

All live migration methods require that the source and destination databases run 
MongoDB 6.0.13+ or MongoDB 7.0.8+. To migrate data from databases using prior 

> versions of MongoDB, see `Legacy Migration <https://www.mongodb.com/docs/atlas/legacy-migration/>`__ or :ref:`arch-center-manual-migration`. 

Reviewer: avoid hard-coded links wherever possible
```suggestion
versions of MongoDB, see :atlas:`Legacy Migration </atlas/legacy-migration/>` or :ref:`arch-center-manual-migration`. 
```

* **Pull data into Atlas.** |service| pulls data from the source MongoDB deployment 
  and requires access to the source deployment through the deployment's firewall. When the 
  {+clusters+} are nearly synced, you must stop write operations on the source, 

>   redirect applications to the |service| {+cluster+} and restart them. The following

Reviewer: ```suggestion
  redirect applications to the |service| {+cluster+}, and restart them. The following
```

  considerations apply:

  - Best for deployments not monitored by |com|. 
  - The source database must be publicly accessible to allow inbound access from the live migration server.
  - Doesn't support :ref:`VPC peering <vpc-peering>` or :ref:`private endpoints <private-endpoint>`
    for either the source or destination {+cluster+}.
  - Source and destination {+cluster+} topologies must match. For example, both 
    must be replica sets or sharded clusters with the same number of shards.
  - Plan for minimal downtime to stop writes and restart applications with a new connection string. 
    The migration process is CPU-intensive and requires significant network bandwidth. MongoDB 

>     recommends that the migration host(s) are run on their own dedicated server with a high 

Reviewer: avoid using (s) to indicate an optional plural https://www.mongodb.com/docs/meta/style-guide/style/plurals/
```suggestion
    recommends that migration hosts are run on their own dedicated server with a high 
```

>     network bandwidth. Provision the migration host(s) with the following minimum configuration:

Reviewer: ```suggestion
    network bandwidth. Provision the migration host with the following minimum configuration:
```

    .. list-table::
       :header-rows: 1
       :widths: 20 40

       * - Number of VMs
         - Total of 3 VMs, 2 for sharded and 1 for RS

       * - Purpose
         - Mongosync

       * - Location
         - Must access on-prem and public cloud

       * - CPU
         - 8

       * - Memory
         - 32 GB

       * - OS
         - OS - 64-bit

       * - Disk Size
         - Disk enough for logging

    To ensure a smooth migration process, confirm that the source cluster's oplog size is adequate to 
    cover the entire migraiton duration.

  For full migration recommendations and instructions, see :ref:`c2c-pull-live-migration`.

> * **Push data into Atlas.** |com| push data to |service| using a secure :term:`link-token` 

Reviewer: ```suggestion
* **Push data into Atlas.** |com| pushes data to |service| using a secure :term:`link-token` 
```

    reflect back on the source.
  - Supports :ref:`VPC peering <vpc-peering>` and :ref:`private endpoints <private-endpoint>`.
  - Source and destination {+cluster+} topologies must match. For example, both 
    must be replica sets or sharded clusters with the same number of shards.
  - Ensure that you connect to your |service| {+cluster+} from all client hardware where your 
    applications run. Testing your connection string helps ensure that your data migration 
    process completes with minimal downtime.
  
  For full migration recommendations and instructions, see :ref:`c2c-push-live-migration`.

Monitoring Migrations
---------------------

.. include:: /includes/cloud-docs/shared-migration-monitoring-description.rst

To learn more, see :ref:`monitor-migrations`.

.. _arch-center-manual-migration:

Manual Migration Methods
------------------------

If |service| live migration can't satisfy the constraints of your migration requirements, 
you can bring data from existing MongoDB deployments, ``JSON``, or ``CSV`` files
into |service| using one of the following tools that you run outside of |service|.

.. include:: /includes/cloud-docs/shared-migration-tools-table.rst

> If you are required to use either |service| VNet peering or Private Link configurations, and/or 

Reviewer: this is a series so no conjunction is needed here
```suggestion
If you are required to use either |service| VNet peering or Private Link configurations, you
```

don't want to allow direct connection from a third party to its source cluster, or if you 
don't already have or don't want to import the source cluster in |onprem| or |mms|, then MongoDB 
recommends the mongosync approach.

If you have relatively small datasets (<300 GB) to migrate, and can afford application downtime 

> for an extended time period, then MongoDB recommends the mongodump and mongorestore approach.

Reviewer: ```suggestion
for an extended time period, then MongoDB recommends the ``mongodump`` and ``mongorestore`` approach.
```
is there anywhere in Atlas docs or server manual where we can link to cover this approach?

If you have relatively small datasets (<300 GB) to migrate, no index concerns, and can afford 

> application downtime for an extended time period, then MongoDB recommends the mongoexport and 
> mongoimport approach.

Reviewer: ```suggestion
application downtime for an extended time period, then MongoDB recommends the ``mongoexport`` and 
``mongoimport`` approach.
```
same comment as above re: links
